src/jit/lower.cpp

```diff
@@ -491,7 +491,63 @@ void Lowering::DecomposeNode(GenTreePtr* pTree, Compiler::fgWalkData* data)
         }
         break;
     case GT_NEG:
-        NYI("GT_NEG of TYP_LONG");
+        {
+            GenTree* op1 = tree->gtGetOp1();
+            noway_assert(op1->OperGet() == GT_LONG);
+
+            // TODO-Cleanup: Identical code exists in DecomposeStoreInd,
+            // it should be moved to a function similar to CreateTemporary
+            GenTreeStmt* newStmt = comp->fgInsertEmbeddedFormTemp(&op1->gtOp.gtOp1);
+            if (newStmt->gtStmtIsTopLevel())
+            {
+                for (GenTreeStmt* nextEmbeddedStmt = newStmt->gtStmtNextIfEmbedded();
+                     nextEmbeddedStmt != nullptr;
+                     nextEmbeddedStmt = nextEmbeddedStmt->gtStmt.gtStmtNextIfEmbedded())
+                {
+                    comp->compCurStmt = nextEmbeddedStmt;
+                    comp->fgWalkTreePost(&nextEmbeddedStmt->gtStmt.gtStmtExpr, Lowering::DecompNodeHelper, this, true);
+                }
+
+                comp->compCurStmt = curStmt;
+            }
+
+            newStmt = comp->fgInsertEmbeddedFormTemp(&op1->gtOp.gtOp2);
+            assert(!newStmt->gtStmtIsTopLevel());  
+
+            GenTree* loOp1 = op1->gtGetOp1();
+            GenTree* hiOp1 = op1->gtGetOp2();
+
+            //
+            // Change the existing GT_NEG node so it computes the high part
+            // and insert the low part computation right before it.
+            //
+
+            GenTree* loNeg = comp->gtNewOperNode(GT_NEG, TYP_INT, loOp1);
+            loNeg->CopyCosts(tree);
+            loNeg->gtFlags = (tree->gtFlags & GTF_LIVENESS_MASK);
```

Comment:
OK, I'll remove `GTF_LIVENESS_MASK` and double check if there's nothing else that needs to be copied. What I know for sure is that I must not copy the side effects. They also need to be removed from the original `GT_NEG` due to the introduction of temporaries.